src/mapper/pkg/resolvers/schema.helpers.resolvers.go

```diff
@@ -378,22 +378,25 @@ func (r *Resolver) handleReportTCPCaptureResults(ctx context.Context, results mo
 	for _, captureItem := range results.Results {
 		logrus.Debugf("Handling TCP capture result from %s to %s:%d", captureItem.SrcIP, captureItem.Destinations[0].Destination, lo.FromPtr(captureItem.Destinations[0].DestinationPort))
 
-		srcSvcIdentity, err := r.discoverInternalSrcIdentity(ctx, captureItem)
-		if errors.Is(err, kubefinder.ErrNoPodFound) {
+		isLocal, err := r.kubeFinder.IsSrcIpLocal(ctx, captureItem.SrcIP)
+		if err != nil {
+			logrus.WithError(err).WithField("ip", captureItem.SrcIP).Error("could not determine if source IP is local")
+			continue
+		}
+		if !isLocal {
 			err := r.reportIncomingInternetTraffic(ctx, captureItem.SrcIP, captureItem.Destinations)
 			if err != nil {
 				logrus.WithError(err).Error("could not report incoming internet traffic")
-				continue
 			}
+			continue
 		}
-
+		srcSvcIdentity, err := r.discoverInternalSrcIdentity(ctx, captureItem)
 		if err != nil {
 			logrus.WithError(err).Debugf("could not discover src identity for '%s'", captureItem.SrcIP)
 			continue
 		}
-
 		for _, dest := range captureItem.Destinations {
-			r.handleExternalIncomingTrafficTCPResult(ctx, srcSvcIdentity, dest)
+			r.handleInternalTrafficTCPResult(ctx, srcSvcIdentity, dest)
```

Comment:
Back to how `isLocal` may sound confusing - here you call it "Internal" traffic, which makes a lot more sense to me. So stick to the "internal/external" phraing (rather than "local" or not).